sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -581,39 +552,44 @@ Member Functions are described in the table below
 a|
 [source,c++]
 ----
-annotated_ref(const annotated_ref&) = default;
+operator T() const;
 ----
 |
-Compiler generated copy constructor.
+Reads the object of type `T` that is referenced by this wrapper, applying the
+annotations when the object is loaded from memory.
 
 // --- ROW BREAK ---
 a|
 [source,c++]
 ----
-operator T() const;
+T operator=(const T &);
 ----
 |
-Reads the object of type `T` that is referenced by this wrapper, applying the
-annotations when the object is loaded from memory.
+Writes an object of type `T` to the location referenced by this wrapper,
+applying the annotations when the object is stored to memory.
 
 // --- ROW BREAK ---
 a|
 [source,c++]
 ----
-annotated_ref& operator=(const T &);
+T operator=(const annotated_ref&);
 ----
 |
 Writes an object of type `T` to the location referenced by this wrapper,
 applying the annotations when the object is stored to memory.
+Does not rebind the reference!
 
 // --- ROW BREAK ---
 a|
 [source,c++]
 ----
-annotated_ref& operator=(const annotated_ref&) = default;
+T operatorOP(const T &);
 ----
 |
-Assign from another `annotated_ref` object.
+Where [code]#OP# is: [code]#pass:[+=]#, [code]#-=#,[code]#*=#, [code]#/=#, [code]#%=#, [code]#+<<=+#, [code]#>>=#, [code]#&=#, [code]#\|=#, [code]#^=#.
+
+Compound assignment operators. Return result by value.
+Available if the corresponding non-compound operator is available for `T`.
```

Comment:
I think we should have some statement here about the annotations because these operations do both a load and a store.  Could we define these operators like this?

> `T operatorOP(const T &val)`
>
> Available only if the corresponding non-compound operator is available for `T`.
> Equivalent to:
> ```
> T tmp = *this;     // Reads from memory while applying annotations
> tmp = tmp OP val;
> *this = tmp;       // Writes to memory while applying annotations
> return tmp;
> ```

For `marray`, these operators are defined as friend functions.  Should we do that here too?

Reply:
Yes making this more explicit is a good idea to avoid confusion.

Overloaded operators should be declared as friend functions if implicit conversion for the left-hand-side should work. This isn't needed here. So it doesn't matter.